utils/gen_pgms_last_commit.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+"""
+Script for creating an pgms_with_last_commit.json file contains all pgms
+with their last commit
+{"pgm": {"commit": "git log -1 PGM src dir"}}.
+
+Used by GitHub actions workflow.
+
+python gen_pgms_last_commit.py src_root_dir
+
+@author Tomas Zigo <tomas.zigo slovanet.sk>
+"""
+
+import json
+import os
+import subprocess
+import shutil
+import sys
+
+
+def get_last_commit(src_dir):
+    """Generate JSON object with the following structure e.g.
+    {"pgm": {"commit": "git log -1 PGM src dir"}}
+
+    :param str src_dir: root src dir
+
+    :return JSON obj result: JSON object
+    """
+    result = {}
+    if not shutil.which("git"):
+        sys.exit("Git command was not found. Please install it.")
+    for root, dirs, files in os.walk(src_dir):
+        for f in files:
+            if f.endswith(".html"):
```

Comment:
Why to limit that to HTML files? We are interested in history of the whole directory, no?

Reply:
It is a directory filter that filters directories that are not core modules based on the assumption that each core module has a manual page in the root directory.

Reply:
Oh, I see, f is not used to get the path. Can you rewrite the code so that the operation on the whole directory is not a inside of a loop over files and add comments? Perhaps something like:

```python
if not contains_html_file(files):
    # No HTML file in the directory, no need to store metadata.
    continue
# ...git here